src/Application.php

```diff
@@ -495,11 +492,16 @@ public function projectWithVersion($project, $version)
      */
     public function buildGlobalCommand($composerCommand, $execPath, $composerArgs, $projectWithVersion, $env, $installLocation)
     {
-        $projectSpecificArgs = array("--working-dir=$installLocation", $composerCommand);
-        if (!empty($projectWithVersion)) {
-            $projectSpecificArgs[] = $projectWithVersion;
+        if ($composerCommand == 'remove') {
```

Comment:
Nice, but you don't need to add special-checking to `buildGlobalCommand()`. If you define a method `removeCommand()`, it will be called instead. You can then build and return whatever sort of `CommandToExec()` you'd like.